builder/builder-next/adapters/containerimage/pull.go

```diff
@@ -114,32 +112,61 @@ func (is *imageSource) resolveLocal(refStr string) ([]byte, error) {
 	return img.RawJSON(), nil
 }
 
-func (is *imageSource) ResolveImageConfig(ctx context.Context, ref string, opt gw.ResolveImageConfigOpt) (digest.Digest, []byte, error) {
-	if preferLocal {
-		dt, err := is.resolveLocal(ref)
-		if err == nil {
-			return "", dt, nil
-		}
-	}
-
+func (is *imageSource) resolveRemote(ctx context.Context, ref string, platform *ocispec.Platform) (digest.Digest, []byte, error) {
 	type t struct {
 		dgst digest.Digest
 		dt   []byte
 	}
 	res, err := is.g.Do(ctx, ref, func(ctx context.Context) (interface{}, error) {
-		dgst, dt, err := imageutil.Config(ctx, ref, is.getResolver(ctx), is.ContentStore, opt.Platform)
+		dgst, dt, err := imageutil.Config(ctx, ref, is.getResolver(ctx), is.ContentStore, platform)
 		if err != nil {
 			return nil, err
 		}
 		return &t{dgst: dgst, dt: dt}, nil
 	})
+	var typed *t
 	if err != nil {
 		return "", nil, err
 	}
-	typed := res.(*t)
+	typed = res.(*t)
 	return typed.dgst, typed.dt, nil
 }
 
+func (is *imageSource) ResolveImageConfig(ctx context.Context, ref string, opt gw.ResolveImageConfigOpt) (digest.Digest, []byte, error) {
+	resolveMode, err := source.ParseImageResolveMode(opt.ResolveMode)
+	if err != nil {
+		return "", nil, err
+	}
+	switch resolveMode {
+	case source.ResolveModeForcePull:
+		dgst, dt, err := is.resolveRemote(ctx, ref, opt.Platform)
+		// TODO: pull should fallback to local in case of failure to allow offline behavior
+		// the fallback doesn't work currently
+		return dgst, dt, err
```

Comment:
nit; could be

```go
return is.resolveRemote(ctx, ref, opt.Platform)
```

Reply:
@thaJeztah true, this was because the original (desired) code is the one commented out below in which case the assignments do make sense.